target/linux/octeon/base-files/lib/upgrade/platform.sh

```diff
@@ -1,123 +1,152 @@
+	return 0
 }
 
 platform_do_upgrade() {
-	local tar_file="$1"
-	local board=$(board_name)
-	local rootfs="$(platform_get_rootfs)"
-	local kernel=
-
-	[ -b "${rootfs}" ] || return 1
-	case "$board" in
-	er)
-		kernel=mmcblk0p1
-		;;
-	erlite)
-		kernel=sda1
-		;;
-	itus,shield-router)
-		kernel=ItusrouterImage
-		;;
-	*)
-		return 1
+	local sysupgrade_file="${1}"
+	local board="$(board_name)"
+	local kernel_file="vmlinux.64"
+	local boot_device
+	local root_device
+
+	case "${board}" in
+		er)
+			boot_device="/dev/mmcblk0p1"
+			root_device="/dev/mmcblk0p2"
+			;;
+		erlite)
+			boot_device="/dev/sda1"
+			root_device="/dev/sda2"
+			;;
+		itus,shield-router)
+			kernel_file="ItusrouterImage"
+			boot_device="/dev/mmcblk1p1"
+			root_device="/dev/mmcblk1p2"
+			;;
+		*)
+			return 1
 	esac
 
-	platform_do_flash $tar_file $board $kernel $rootfs
+	# checks if:
+	# 1. boot device variable is not empty
+	# 2. boot device variable points to actual boot device
+	# 3. root device variable is not empty
+	# 4. root device variable points to actual root device
+	# and flashes firmware
+	if [ -n "${boot_device}" ] && \
+		[ -b "${boot_device}" ] && \
+		[ -n "${root_device}" ] && \
+		[ -b "${root_device}" ] ; then
+		platform_do_flash "${sysupgrade_file}" "${kernel_file}" "${boot_device}" "${root_device}"
+	fi
 
 	return 0
 }
 
 platform_check_image() {
-	local board=$(board_name)
-	local tar_file="$1"
-
-	local board_dir=$(tar tf "$tar_file" | grep -m 1 '^sysupgrade-.*/$')
-	board_dir=${board_dir%/}
-	[ -n "$board_dir" ] || return 1
-
-	case "$board" in
-	er | \
-	erlite | \
-	itus,shield-router)
-		local kernel_length=$(tar xf $tar_file $board_dir/kernel -O | wc -c 2> /dev/null)
-		local rootfs_length=$(tar xf $tar_file $board_dir/root -O | wc -c 2> /dev/null)
-		[ "$kernel_length" = 0 -o "$rootfs_length" = 0 ] && {
-			echo "The upgrade image is corrupt."
-			return 1
-		}
-		return 0
-		;;
+	local sysupgrade_file="${1}"
+	local board="$(board_name)"
+	# find a sysupgrade directory by looking inside sysupgrade file
+	local sysupgrade_directory="$(tar tf ${sysupgrade_file} | awk -F '/' '/^sysupgrade/ {print $1; exit}')"
+	# checks if sysupgrade directory variable is not empty and exits with code 1 otherwise
+	[ -n "${sysupgrade_directory}" ] || return 1
+
+	case "${board}" in
+		er | \
+		erlite | \
+		itus,shield-router)
+			# checks if:
+			# 1. sysupgrade kernel file length is 0
+			# 2. sysupgrade rootfs file length is 0
+			# and exits with code 1
+			local kernel_length="$(tar xf ${sysupgrade_file} ${sysupgrade_directory}/kernel -O | wc -c 2> /dev/null)"
+			local rootfs_length="$(tar xf ${sysupgrade_file} ${sysupgrade_directory}/root -O | wc -c 2> /dev/null)"
+			if [ "${kernel_length}" = 0 ] || [ "${rootfs_length}" = 0 ] ; then
+				echo "The upgrade image is corrupt."
+				return 1
+			fi
+			return 0
+			;;
 	esac
 
-	echo "Sysupgrade is not yet supported on $board."
+	echo "Sysupgrade is not yet supported on ${board}."
```

Comment:
see above.

Reply:
should i use raw variables without curly brackets?

there does not seem to be a code guideline and it would preffered to keep the same style.